tools/docker/bazel.dockerfile

```diff
@@ -0,0 +1,13 @@
+ARG BASE_IMAGE_OS_NAME=fedora
+ARG BASE_IMAGE_OS_VERSION=38
+FROM ${BASE_IMAGE_OS_NAME}:${BASE_IMAGE_OS_VERSION}
+
+COPY --chown=0:0 bazel/install-deps.sh /
+
+RUN dnf install -y wget || { apt-get update && apt install -y wget; }
+RUN wget -O /usr/local/bin/bazel \
+        https://github.com/bazelbuild/bazelisk/releases/latest/download/bazelisk-linux-amd64 && \
+        chmod +x /usr/local/bin/bazel
+
+# run after wget installation to take advantage of pkg cache cleaning
```

Comment:
If you want to optimize for image size then all the lines (from dnf install to this one) need to be merged together. Each RUN creates a new image layer. Cleaning package cache at this point doesn't achieve much since it is present in the layer created on line 7.

Move bazelisk installation to install-deps behind a switch as well?

Reply:
i think we can move bazelisk installation into `tools/bazel`. i'd rather not be too opinionated about how someone installs deps on the system.

Reply:
ahh right. yeh i think there is also an option for squashing all the layers together? ill take a look. it's not too big either way.